### DISCUSSION TITLE
How to use the tftp command in shell?

### DISCUSSION THREAD

**Discussion by ayushdas2000** [Kostr/UEFI-Lessons#6] [General]
Hi everyone.I have been trying to download a file using tftp. When i use the command tftp -i [ip-address] get filename i get that 'tftp' is not recognized as an internal or external command, operable program, or script file.Do i need to configure or load something before this. Any leads on how to proceed with tftp will be appreciated. Thanks . (I am also open to using https if that is a better option)

**Comment by Kostr** [comment]
You can build it as a standalone application https://github.com/tianocore/edk2/blob/master/ShellPkg/DynamicCommand/TftpDynamicCommand/TftpApp.inf

**Reply by ayushdas2000** [reply]
Thanks Kostr. I have the tftp.efi application. When i use the command tftp i get the message no network interface found. Also ifconig -l lists nothing. Any idea what should be done to use the networking features?

**Reply by Kostr** [reply]
You need to tell more about your hardware. Apparently your PCIe network device didn't produce network protocols. So either it doesn't have UEFI OptionROM, or BIOS didn't read it, because of some policy enabled in its settings.

**Reply by ayushdas2000** [reply]
Thanks Kostr. You were right about the settings. I had to enable Network book and now ifconfig works. However it only lists eth0(media disconnected) Is there a way i could get wirless interface here. Thanks.

**Reply by ayushdas2000** [reply]
Also , Do i need to load some drivers to be able to use the networking stack ?

**Reply by ayushdas2000** [reply]
Fixed this up. Will sum it up for anyone facing the same issues:
1. Build tftp.efi from tianocore.
2. Enable network stack from bios setup.
3. Use ifconfig to setup ip for your network interface.
4. setup your tftp server.
5. use tftp ip-addr filename to download the file
This link is useful : https://superuser.com/questions/1179084/copying-files-to-efi-shell-through-network
Note : you might have to use connect -r before configuring your network interface.
